### PR TITLE
8 transactions middleware

### LINKED ISSUE #8: Accept transactions from the frontend and pass it to mining service
**Description:**

- When a user makes a payment, the transaction should be passed on to the blockchain to be added.

**Acceptance Criteria:**

- [x] The transactions are verified before being passed on to the blockchain.
- [x] The transactions are successfully added or denied on failures.
- [x] The front-end is notified of the results.

### PR DESCRIPTION
**Pull request by abhisachdev17**
## Summary of Changes
- The transaction middleware that temporarily has the transaction queue and all the private keys
- Once the user and wallet services are moved those will move to appropriate places
- A lot of tests to test the routes and individual function
- Shared module that has the HttpCode file
- Replace test_models with test_functions as there were no models to test

Closes #8

### CODE REVIEW

File: backend/services/transactions/src/routes.py [hunk 1]
```diff
@@ -1,5 +1,127 @@
...
+INCORRECT_PAYLOAD_MSG = "Please send correct json payload"
+VERFICATION_FAILURE= "Unable to Verify the Transaction"
+
+## Temporary
+TransactionPool = []
+private_keys = {}
+public_keys = {}
+
+
+def validateSignature(id, signature, address):
```
Comment (ZackHolmberg) [inline backend/services/transactions/src/routes.py:15]: Keep this and createTransaction() minus its key-related stuff and take out all other endpoints

File: backend/services/transactions/tests/functional/test_routes.py [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import pytest
```
Comment (ZackHolmberg) [inline backend/services/transactions/tests/functional/test_routes.py:1]: Can clean up/remove a lot of this after routes update

**Review by ZackHolmberg — CHANGES_REQUESTED** [review]

File: backend/services/transactions/src/routes.py [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+        isVerified = validateSignature(transaction_id, signature, from_address)
+
+    except Exception as e:
+        return jsonify(err= str(e)), HttpCode.BAD_REQUEST
+
+
+    if not isVerified:
+        return jsonify(err= FailureReturnString.TRANSACTION_VERFICATION_FAILURE), HttpCode.UNAUTHORIZED
+    
+    # call wallet to confirm if the amount event exists
```
Comment (ZackHolmberg) [inline backend/services/transactions/src/routes.py]: Nitpick but can you add `TODO:` to the start of this comment and the below one please

**Review by ZackHolmberg — APPROVED** [review]

**Review by madisonfines — APPROVED** [review]

File: backend/services/transactions/src/routes.py (hunk 3 above)
Comment (abhisachdev17) [reply backend/services/transactions/src/routes.py]: Done!

**Pull request merged**